src/auxil.c

```diff
@@ -81,55 +81,65 @@ void set_rho_vec(OSQPWorkspace *work) {
 
   work->settings->rho = c_min(c_max(work->settings->rho, RHO_MIN), RHO_MAX);
 
+  c_float* l           = OSQPVectorf_data(work->data->l);
+  c_float* u           = OSQPVectorf_data(work->data->u);
+  c_float* rho_vec     = OSQPVectorf_data(work->rho_vec);
+  c_int   *constr_type = OSQPVectori_data(work->constr_type);
+
   for (i = 0; i < work->data->m; i++) {
```

Comment:
We can similarly define `c_int m = work->data->m` and use it in the loop condition.

Reply:
fixed.